src/base/memory_monitor_sr.cpp

```diff
@@ -113,23 +113,23 @@ namespace cubmem
     metainfo.allocated_size = (uint64_t) size;
     m_total_mem_usage += metainfo.allocated_size;
 
-    tag_name = make_tag_name (file, line);
+    stat_name = make_stat_name (file, line);
 
-    std::unique_lock <std::mutex> map_lock (m_map_mutex);
-    auto search = m_tag_map.find (tag_name);
-    if (search != m_tag_map.end ())
+    std::unique_lock <std::shared_mutex> stat_map_write_lock (m_stat_map_mutex);
+    auto search = m_stat_name_map.find (stat_name);
```

Comment:
```suggestion
    const auto search = m_stat_name_map.find (stat_name);
```

if there is no intention to modify `search`

Reply:
Done.